src/core_plugins/kibana/public/dashboard/panel/panel.js

```diff
@@ -72,116 +66,69 @@ uiModules
-        } else if ($scope.panel.type === savedSearches.type) {
-          if ($scope.savedObj.searchSource) {
-            $scope.registerPanelIndexPattern($scope.panel.panelIndex, $scope.savedObj.searchSource.get('index'));
-          }
-          // This causes changes to a saved search to be hidden, but also allows
-          // the user to locally modify and save changes to a saved search only in a dashboard.
-          // See https://github.com/elastic/kibana/issues/9523 for more details.
-          $scope.panel.columns = $scope.panel.columns || $scope.savedObj.columns;
-          $scope.panel.sort = $scope.panel.sort || $scope.savedObj.sort;
+      const containerAPI = {
+        savePanelState,
+        getAppState: () => $scope.appState,
+        getIsViewOnlyMode: $scope.isViewOnlyMode,
+        createChildUiState: $scope.createChildUiState,
+        registerPanelIndexPattern: $scope.registerPanelIndexPattern,
+      };
 
-          $scope.setSortOrder = function setSortOrder(columnName, direction) {
-            $scope.panel.sort = [columnName, direction];
-            $scope.saveState();
-          };
+      const handleError = (error) => {
+        $scope.error = error.message;
 
-          $scope.addColumn = function addColumn(columnName) {
-            $scope.savedObj.searchSource.get('index').popularizeField(columnName, 1);
-            columnActions.addColumn($scope.panel.columns, columnName);
-            $scope.saveState();  // sync to sharing url
-          };
+        // Dashboard listens for this broadcast, once for every visualization (pendingVisCount).
+        // We need to broadcast even in the event of an error or it'll never fetch the data for
+        // other visualizations.
+        $scope.$root.$broadcast('ready:vis');
 
-          $scope.removeColumn = function removeColumn(columnName) {
-            $scope.savedObj.searchSource.get('index').popularizeField(columnName, 1);
-            columnActions.removeColumn($scope.panel.columns, columnName);
-            $scope.saveState();  // sync to sharing url
-          };
+        // If the savedObjectType matches the panel type, this means the object itself has been deleted,
+        // so we shouldn't even have an edit link. If they don't match, it means something else is wrong
+        // with the object (but the object still exists), so we link to the object editor instead.
+        const objectItselfDeleted = error.savedObjectType === $scope.panel.type;
+        if (objectItselfDeleted) return;
 
-          $scope.moveColumn = function moveColumn(columnName, newIndex) {
-            columnActions.moveColumn($scope.panel.columns, columnName, newIndex);
-            $scope.saveState();  // sync to sharing url
-          };
-        }
+        const type = $scope.panel.type;
+        const id = $scope.panel.id;
+        const service = _.find(services, { type: type });
```

Comment:
nitpick (and I know it's just copy/pasted from the old code): If we're still on the "use less lodash", this can be `services.find(service => service.type === type)`.